src/prettytable/prettytable.py

```diff
@@ -183,9 +208,14 @@ def __init__(self, field_names=None, **kwargs):
         self._left_padding_width = kwargs["left_padding_width"] or None
         self._right_padding_width = kwargs["right_padding_width"] or None
 
-        self._vertical_char = kwargs["vertical_char"] or "|"
-        self._horizontal_char = kwargs["horizontal_char"] or "-"
-        self._junction_char = kwargs["junction_char"] or "+"
+        # color codes to theme color, then back to base
+        # _theme["border"] + kwargs["char"] + _theme["base"]
+        self._theme = kwargs["theme"] or THEME.DEFAULT
+        # Converted to try except switch because the "or" keyword wasn't working
+        self._raw_vertical_char = kwargs["vertical_char"] or "|"
+        self._raw_horizontal_char = kwargs["horizontal_char"] or "-"
+        self._raw_junction_char = kwargs["junction_char"] or "+"
+        self.set_theme(kwargs["theme"] or THEME.DEFAULT)
```

Comment:
Maybe some of the test failures are coming because we're setting a default theme here, and `set_theme` then inserts some colour reset codes.

One idea: for the no-colour case, instead of setting a default theme with `""` for the colours, ditch a default theme and set `self._theme = None` to begin with.

And then only set, for example, `self._vertical_char` values inside `set_theme`, when actually called to set a theme. (Or perhaps check if a theme is set when returning a value in `vertical_char()`.)

Reply:
Fixed